### PR TITLE
[FR] Add default log options to training client and test

### PR DESCRIPTION
**Pull request by mssfang**
Added getDefaultLogOptions method to training client builder. 

A follow up PR for https://github.com/Azure/azure-sdk-for-java/pull/20883#discussion_r618522479

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClientBuilder.java [hunk 1]
```diff
@@ -207,6 +208,15 @@ public FormTrainingClientBuilder httpLogOptions(HttpLogOptions logOptions) {
         return this;
     }
 
+    /**
+     * Gets the default Azure Form Training client headers and query parameters allow list.
+     *
+     * @return The default {@link HttpLogOptions} allow list.
+     */
+    public static HttpLogOptions getDefaultLogOptions() {
```
Comment (srnagar) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClientBuilder.java:216]: It's good to include the reason for adding this API in the PR description. Why do we want to add this API?

Comment (mssfang) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClientBuilder.java:216]: It is better to have default log options methods. @alzimmermsft

**Review by alzimmermsft — APPROVED** [review]

**Review by kristapratico — APPROVED** [review]
FR part LGTM

**Review by maririos — APPROVED** [review]

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClientBuilder.java [hunk 2]
```diff
@@ -207,6 +208,15 @@ public FormTrainingClientBuilder httpLogOptions(HttpLogOptions logOptions) {
         return this;
     }
 
+    /**
+     * Gets the default Azure Form Training client headers and query parameters allow list.
```
Comment (srnagar) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClientBuilder.java:212]:
```suggestion
     * Gets the default Azure Form Training client headers and query parameters that are logged by default if HTTP logging is enabled.
```

**Review by srnagar — APPROVED** [review]

**Pull request merged**